Content.Client/GameObjects/Components/Strap/StrapVisualizer.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Content.Shared.GameObjects.Components.Strap;
+using JetBrains.Annotations;
+using Robust.Client.GameObjects;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Client.GameObjects.Components.Strap
+{
+    /// <summary>
+    /// This manages an object with a <see cref="StrapComponent"/> visuals.
+    /// You specify a buckledSuffix in the prototype, and the visualizer will look for
+    /// [state]_buckledSuffix state for every layer in the sprite.
+    /// If it doesn't find it, it leaves the state as-is.
+    ///
+    /// By default this suffix is _buckled.
+    /// </summary>
+    [UsedImplicitly]
+    public class StrapVisualizer : AppearanceVisualizer
+    {
+        /// <summary>
+        /// The suffix used to define the buckled states names in RSI file.
+        /// [state] becomes [state][suffix],
+        /// default suffix is _buckled, so chair becomes chair_buckled
+        /// </summary>
+        [ViewVariables][DataField("buckledSuffix", required: false)]
+        private string? _buckledSuffix = "_buckled";
+
+        // This array keeps the original layers' states of the sprite
+        private string?[]? _defaultStates;
+
+        // This keeps a reference to the entity's SpriteComponent to avoid fetching it each time
+        private ISpriteComponent? _sprite;
+
+        private bool? _isBuckled;
```

Comment:
As above on flyweighting.

Also if you're gonna store it use a ComponentDependency.

Reply:
did someone add compdeps to visualizers? i didn't

Reply:
As you stated, I shouldn't cache Sprite here, I was already told on Discord by someone else but I forgot that I didn't change it